include/picongpu/fields/MaxwellSolver/YeePML/YeePML.kernel

```diff
@@ -0,0 +1,587 @@
+            DataSpace< simDim > const numGuardCells
+        ):
+            Parameters( parameters ),
+            negativeBorderSize( precisionCast< float_X >( localThickness.negativeBorder ) ),
+            positiveBorderSize( precisionCast< float_X >( localThickness.positiveBorder ) ),
+            numLocalDomainCells( numLocalDomainCells ),
+            numGuardCells( numGuardCells )
+        {
+        }
+    };
+
+    namespace detail
+    {
+
+        /** Get relative depth of a given cell for 1D.
+         *
+         * This function operates with a 1D slice of domain and PML.
+         * index == numGuardCells corresponds to the external negative PML
+         * border, and index == numLocalDomainCells - numGuardCells - 1
+         * corresponds to the external positive PML border.
+         * For the internal area result is 0, for points in PML the depth
+         * scales from 0 at the internal border to 1 at the external border.
+         * Index and local domain size include the guard.
+         *
+         * @param cellIdx cell index including the guard, can be fractional,
+         * e.g. for halves of cells
+         * @param numPMLCellsNegative number of PML cells at the negative border
+         * @param numPMLCellsPositive number of PML cells at the positive border
+         * @param numLocalDomainCells number of cells of the local domain
+         * including the guard
+         * @param numGuardCells number of guard cells at each side
+         * @return relative depth, value between 0 and 1
+         */
+        DINLINE float_X getRelativeDepth(
+            float_X const cellIdx,
+            float_X const numPMLCellsNegative,
+            float_X const numPMLCellsPositive,
+            uint32_t const numLocalDomainCells,
+            uint32_t const numGuardCells
+        )
+        {
+            auto zeroBasedIdx = cellIdx - numGuardCells;
+            auto const isInLeftPML = ( zeroBasedIdx < numPMLCellsNegative );
+            if( isInLeftPML )
+                return ( numPMLCellsNegative - zeroBasedIdx ) / numPMLCellsNegative;
+            else
+            {
+                auto zeroBasedRightPMLStart = numLocalDomainCells -
+                    2 * numGuardCells - numPMLCellsPositive;
+                auto const isInRightPML = ( zeroBasedIdx > zeroBasedRightPMLStart );
+                if( isInRightPML )
+                    return ( zeroBasedIdx - zeroBasedRightPMLStart ) / numPMLCellsPositive;
+            }
+            return 0._X;
+        }
+
+        /** Get absorption parameters: sigma, kappa and alpha at a given cell
+         *
+         * Apply polynomial grading, as described in pml.param.
+         *
+         * @param cellIdx cell index including the guard, can be fractional,
+         * e.g. for halves of cells
+         * @param parameters parameters of PML in the local domain
+         * @param normalizedSigma value of normalized sigma at the cell, output
```

Comment:
Why do you add `, output`? 
Same applies to two lines below.

Reply:
Will change these to the doxygen `@param[out]`.